packages/flutter_tools/lib/src/commands/upgrade.dart

```diff
@@ -337,7 +269,18 @@ class UpgradeCommandRunner {
         throwToolExit(errorString);
       }
     }
-    verifyStandardRemote(localVersion);
+    // At this point the current checkout should be on HEAD of a branch having
+    // an upstream. Check whether this upstream is "standard".
+    try {
+      VersionUpstreamValidator(version: localVersion, platform: globals.platform).run();
+    } on VersionCheckError catch (error) {
+      throwToolExit(
```

Comment:
I think it's a little unusual for the API of this method to be `void` while it is the responsibility of the caller to catch a `VersionCheckError`. Could you instead have this method return `VersionCheckError?`

Reply:
Done.